corehq/apps/hqwebapp/views.py

```diff
@@ -378,16 +382,9 @@ def _login(req, domain_name):
     req.base_template = settings.BASE_TEMPLATE
 
     context = {}
-    template_name = 'login_and_password/login.html'
-    custom_landing_page = settings.CUSTOM_LANDING_TEMPLATE
-    if custom_landing_page:
-        if isinstance(custom_landing_page, str):
-            template_name = custom_landing_page
-        else:
-            template_name = custom_landing_page.get(req.get_host())
-            if template_name is None:
-                template_name = custom_landing_page.get('default', template_name)
-    elif domain_name:
+    custom_login_page = get_custom_login_page(req)
+    template_name = custom_login_page if custom_login_page else 'login_and_password/login.html'
+    if not custom_login_page and domain_name:
```

Comment:
(copying commit message) 

this is a small change in behavior that I think is more correct than previous. now the domain-specific logic will run if no custom login page is *matched*. previously it would only run of no custom login page was *configured*. however, it was impossible for it to be configured without matching because it would have 500'd from trying to render a template that was `None`